scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,862 @@
+            throw e;
+        } catch (Exception e) {
+            throw new ClosureEvaluationException(EXCEPTION_WHEN_EVALUATING_CLOSURE + StackTraceUtil.getStackTrace(e));
+        }
+        return answer;
+    }
+
+    @Override
+    public PredicateActionResult conditionalCompute(String originator, TaskId taskid, String channel, String key,
+            String predicate, String thenRemappingFunction, String elseRemappingFunction)
+            throws InvalidChannelException, CompilationException, IOException {
+        PredicateActionResult answer;
+        try {
+            Channel chosenChannel = getChannel(channel);
+            Serializable value = chosenChannel.get(key);
+            if (evaluateClosure(predicate, BiPredicate.class).test(key, value)) {
+                answer = new PredicateActionResult(true,
+                                                   chosenChannel.compute(key,
+                                                                         (BiFunction<? super String, ? super Serializable, ? extends Serializable>) evaluateClosure(thenRemappingFunction,
+                                                                                                                                                                    BiFunction.class)));
+            } else {
+                answer = new PredicateActionResult(false,
+                                                   chosenChannel.compute(key,
+                                                                         (BiFunction<? super String, ? super Serializable, ? extends Serializable>) evaluateClosure(elseRemappingFunction,
+                                                                                                                                                                    BiFunction.class)));
+            }
+            commitIfNeeded(channel);
+            logWithContext(taskid,
+                           channel,
+                           "Run conditionalCompute" + ON_KEY + QUOTE + key + QUOTE + RESULT_IS + answer,
+                           Level.DEBUG);
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new ClosureEvaluationException(EXCEPTION_WHEN_EVALUATING_CLOSURE + StackTraceUtil.getStackTrace(e));
+        }
+        return answer;
+    }
+
+    @Override
+    public void conditionalReplaceAll(String originator, TaskId taskid, String channel, String predicate,
+            String thenRemappingFunction) throws InvalidChannelException, CompilationException, IOException {
+        String currentKey = null;
+        try {
+            Channel chosenChannel = getChannel(channel);
+            for (Map.Entry<String, Serializable> entry : chosenChannel.entrySet()) {
+                currentKey = entry.getKey();
+                Serializable value = chosenChannel.get(currentKey);
```

Comment:
Why not getting the value with `entry.getValue()`?

Reply:
yes you are right